bc_obps/registration/tests/test_endpoints.py

```diff
@@ -0,0 +1,187 @@
+from registration.models import NaicsCode, NaicsCategory, Operation, Operator
+from django.test import TestCase
+from localflavor.ca.models import CAPostalCodeField
+from registration.models import NaicsCode, NaicsCategory, Document, Contact, Operation, Operator
+
+pytestmark = pytest.mark.django_db
+
+# initialize the APIClient app
+client = Client()
+
+base_endpoint = '/api/registration/'
+
+
+def mock_postal_code():
+    return 'v8v3g1'
+
+
+baker.generators.add(CAPostalCodeField, mock_postal_code)
+
+
+class TestNaicsCodeEndpoint:
+    endpoint = base_endpoint + 'naics_codes'
+
+    def test_get_method_for_200_status(self, client):
+        response = client.get(self.endpoint)
+        assert response.status_code == 200
+
+    def test_get_method_with_mock_data(self, client):
+        baker.make(NaicsCode, _quantity=2)
+
+        response = client.get(self.endpoint)
+        assert response.status_code == 200
+        assert len(json.loads(response.content)) == 2
+
+
+class TestNaicsCategoriesEndpoint:
+    endpoint = base_endpoint + 'naics_categories'
+
+    def test_get_method_for_200_status(self, client):
+        response = client.get(self.endpoint)
+        assert response.status_code == 200
+
+    def test_get_method_with_mock_data(self, client):
+        baker.make(NaicsCategory, _quantity=3)
+
+        response = client.get(self.endpoint)
+        assert response.status_code == 200
+        assert len(json.loads(response.content)) == 3
+
+
+class TestOperationsEndpoint:
+    endpoint = base_endpoint + 'operations'
+
+    def test_get_method_for_200_status(self, client):
+        response = client.get(self.endpoint)
+        assert response.status_code == 200
+
+    def test_get_method_with_mock_data(self, client):
+        baker.make(Operation, _quantity=1)
+        response = client.get(self.endpoint)
+        assert response.status_code == 200
+        assert len(json.loads(response.content)) == 1
+
+    def test_post_new_operation(self, client):
+        naics_code = baker.make(NaicsCode)
+        naics_category = baker.make(NaicsCategory)
+        document = baker.make(Document)
+        contact = baker.make(Contact, postal_code='V1V 1V1')
+        operator = baker.make(Operator)
+        mock_operation = OperationIn(
+            name='Springfield Nucular Power Plant',
+            type='Type 1',
+            naics_code_id=naics_code.id,
+            naics_category_id=naics_category.id,
+            reporting_activities='donuts eaten',
+            physical_street_address='19 Evergreen Terrace',
+            physical_municipality='Springfield',
+            physical_province='BC',
+            physical_postal_code='V1V 1V1',
+            legal_land_description='It\'s legal',
+            latitude=90,
+            longitude=-120,
+            petrinex_ids=['123', '124'],
+            regulated_products=[1, 2],
+            documents=[document.id],
+            contacts=[contact.id],
+            status='Pending',
```

Comment:
I think that the status field should not be part of the OperationIn class - it introduces the opportunity for an applicant to set the status to 'Approved' automatically, which we don't want.

Instead we should submit the mock_operation body with no set status, then once it's been added, fetch that same operation with the get-by-id operation endpoint, and check then that the returned operation now has a status of Pending

(This will also affect your unit tests for the PUT request)

Reply:
The put request doesn't return the full operation, just name, so I didn't add it there (but maybe put should be returning everything?)

Reply:
oh I just meant removing the status field from the OperationIn object used for the put test. You're correct that we may need to change the response sent from the PUT request later, but until we know for sure, let's leave it as is for now :)